example/generator/odp_generator.c

```diff
@@ -811,21 +809,14 @@ static int gen_recv_thread(void *arg)
 			continue;
 		for (i = 0, pkt_cnt = 0; i < ev_cnt; i++) {
 			pkt = odp_packet_from_event(events[i]);
-			itf = &itfs[odp_pktio_index(odp_packet_input(pkt))];
 
-			if (odp_packet_has_ipv4(pkt)) {
-				if (itf->config.pktin.bit.ipv4_chksum) {
-					if (odp_packet_has_l3_error(pkt))
-						printf("HW detected L3 error\n");
-				}
-			}
+			csum_status = odp_packet_l3_chksum_status(pkt);
+			if (csum_status == ODP_PACKET_CHKSUM_BAD)
+				printf("L3 checksum error detected.\n");
```

Comment:
prints for fast path functions is bad thing. Bellow is print_pkts() which  does all printing. Or we need to move it here at least. Or better to move out all printing from fast path.

Reply:
Prints were already there. We only changed the conditions for printing to highlight the new API.